### PR TITLE
docs: api/grn_table: move grn_table_delete_by_id() reference to the header file

### PR DESCRIPTION
**Pull request by abetomo**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/table.h [hunk 1]
```diff
@@ -165,6 +165,17 @@ grn_table_delete(grn_ctx *ctx,
...
+/**
+ * \brief Delete the record matching the id in the table.
+ *
+ * \param ctx The context object
+ * \param table The table
+ * \param id Id of record to be deleted
+ *
+ * \return \ref GRN_SUCCESS on success, \ref GRN_INVALID_ARGUMENT if the
+ *         record does not exist, and the appropriate \ref grn_rc for any
+ *         other errors.
```
Comment (otegami) [inline include/groonga/table.h:177]: How about using `A on success, B on error` format for consistency?

```suggestion
 * \return \ref GRN_SUCCESS on success, the appropriate \ref grn_rc on error.
 *         For example, \ref GRN_INVALID_ARGUMENT is returned if the
 *         record does not exist.
```

**Review by otegami — APPROVED** [review]
LGTM 👍🏾

**Review by komainu8 — APPROVED** [review]

File: include/groonga/table.h [hunk 2]
```diff
@@ -165,6 +165,17 @@ grn_table_delete(grn_ctx *ctx,
                  grn_obj *table,
                  const void *key,
                  unsigned int key_size);
+/**
+ * \brief Delete the record matching the id in the table.
```
Comment (kou) [inline include/groonga/table.h:169]:
```suggestion
 * \brief Delete the record matching the ID in the table.
```

File: include/groonga/table.h [hunk 3]
```diff
@@ -165,6 +165,17 @@ grn_table_delete(grn_ctx *ctx,
...
+ *
+ * \param ctx The context object
+ * \param table The table
+ * \param id Id of record to be deleted
```
Comment (kou) [inline include/groonga/table.h:173]:
```suggestion
 * \param id ID of record to be deleted
```

**Review by otegami — APPROVED** [review]

**Pull request merged**